test/constructors.jl

```diff
@@ -34,39 +34,68 @@ const ≅ = isequal
                               CategoricalVector{Union{Float64, Missing}}(ones(3))])
     @test df == DataFrame(Any[CategoricalVector{Union{Float64, Missing}}(zeros(3)),
                               CategoricalVector{Union{Float64, Missing}}(ones(3))], [:x1, :x2])
+    @test df == DataFrame(Any[CategoricalVector{Union{Float64, Missing}}(zeros(3)),
+                              CategoricalVector{Union{Float64, Missing}}(ones(3))], ["x1", "x2"])
     @test df == DataFrame(AbstractVector[CategoricalVector{Union{Float64, Missing}}(zeros(3)),
                                          CategoricalVector{Union{Float64, Missing}}(ones(3))], [:x1, :x2])
+    @test df == DataFrame(AbstractVector[CategoricalVector{Union{Float64, Missing}}(zeros(3)),
+                                         CategoricalVector{Union{Float64, Missing}}(ones(3))], ["x1", "x2"])
     @test df == DataFrame((CategoricalVector{Union{Float64, Missing}}(zeros(3)),
                            CategoricalVector{Union{Float64, Missing}}(ones(3))))
     @test df == DataFrame((CategoricalVector{Union{Float64, Missing}}(zeros(3)),
                            CategoricalVector{Union{Float64, Missing}}(ones(3))), (:x1, :x2))
+    @test df == DataFrame((CategoricalVector{Union{Float64, Missing}}(zeros(3)),
+                           CategoricalVector{Union{Float64, Missing}}(ones(3))), (:x1, :x2))
+    @test df == DataFrame((CategoricalVector{Union{Float64, Missing}}(zeros(3)),
+                           CategoricalVector{Union{Float64, Missing}}(ones(3))), ("x1", "x2"))
+    @test df == DataFrame((CategoricalVector{Union{Float64, Missing}}(zeros(3)),
+                           CategoricalVector{Union{Float64, Missing}}(ones(3))), ("x1", "x2"))
     @test df == DataFrame(x1 = Union{Int, Missing}[0.0, 0.0, 0.0],
                           x2 = Union{Int, Missing}[1.0, 1.0, 1.0])
     @test df == DataFrame!(x1 = Union{Int, Missing}[0.0, 0.0, 0.0],
                            x2 = Union{Int, Missing}[1.0, 1.0, 1.0])
     @test df == DataFrame([:x1=>Union{Int, Missing}[0.0, 0.0, 0.0],
                            :x2=>Union{Int, Missing}[1.0, 1.0, 1.0]])
+    @test df == DataFrame([:x1=>Union{Int, Missing}[0.0, 0.0, 0.0],
+                           :x2=>Union{Int, Missing}[1.0, 1.0, 1.0]])
+    @test df == DataFrame!((:x1=>Union{Int, Missing}[0.0, 0.0, 0.0],
+                            :x2=>Union{Int, Missing}[1.0, 1.0, 1.0]))
     @test df == DataFrame!((:x1=>Union{Int, Missing}[0.0, 0.0, 0.0],
                             :x2=>Union{Int, Missing}[1.0, 1.0, 1.0]))
+    @test df == DataFrame(["x1"=>Union{Int, Missing}[0.0, 0.0, 0.0],
+                           "x2"=>Union{Int, Missing}[1.0, 1.0, 1.0]])
+    @test df == DataFrame(["x1"=>Union{Int, Missing}[0.0, 0.0, 0.0],
+                           "x2"=>Union{Int, Missing}[1.0, 1.0, 1.0]])
+    @test df == DataFrame!(("x1"=>Union{Int, Missing}[0.0, 0.0, 0.0],
+                            "x2"=>Union{Int, Missing}[1.0, 1.0, 1.0]))
```

Comment:
Also here:
```suggestion
```

Reply:
they are not duplicated but I put `!` in a wrong place - I will fix it.